cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HTableNameConverter.java

```diff
@@ -67,38 +66,33 @@ private String getBackwardCompatibleTableName(String tablePrefix, TableId tableI
    */
   protected String getHBaseTableName(String tablePrefix, TableId tableId) {
     Preconditions.checkArgument(tablePrefix != null, "Table prefix should not be null.");
-    return getHBaseTableName(getBackwardCompatibleTableName(tablePrefix, tableId));
+    return encodeHBaseEntity(getBackwardCompatibleTableName(tablePrefix, tableId));
   }
 
   /**
    * Gets the system configuration table prefix.
    *
-   * @param htd Table descriptor for any table that can be associated with the
+   * @param prefix Prefix string
    * @return System configuration table prefix (full table name minus the table qualifier).
    * Example input: "cdap_ns.table.name"  -->  output: "cdap_system."   (hbase 94)
    * Example input: "cdap.table.name"     -->  output: "cdap_system."   (hbase 94. input table is in default namespace)
    * Example input: "cdap_ns:table.name"  -->  output: "cdap_system:"   (hbase 96, 98)
    */
-  public abstract String getSysConfigTablePrefix(HTableDescriptor htd);
+  public String getSysConfigTablePrefix(String prefix) {
+    return prefix + "_" + Id.Namespace.SYSTEM.getId() + ":";
```

Comment:
may be. All the classes that implemented this method had the same logic. hence, moved it up.